src/main/java/org/sagebionetworks/bridge/upload/TranscribeConsentHandler.java

```diff
@@ -66,15 +68,28 @@ public void handle(@Nonnull UploadValidationContext context) {
             RequestContext.acquireAccountIdentity(account);
             
             DateTime studyStartTime = participantService.getStudyStartTime(account);
-            LocalDate studyStartDate = studyStartTime.withZone(BridgeConstants.LOCAL_TIME_ZONE).toLocalDate();
-            LocalDate todayLocalDate = DateUtils.getCurrentCalendarDateInLocalTime();
-
-            // "dayInStudy" is 1-indexed. The first day of the study is day 1. Therefore, add 1.
-            int dayInStudy = Days.daysBetween(studyStartDate, todayLocalDate).getDays() + 1;
+            int dayInStudy = calculateDaysSince(studyStartTime);
             record.setDayInStudy(dayInStudy);
+            
+            Map<String, DateTime> studyStartTimeMap = participantService.getStartTimeInEachStudy(account);
+            Map<String, Integer> dayInStudyMap = new HashMap<>();
+            for(Map.Entry<String, DateTime> oneStartTime : studyStartTimeMap.entrySet()) {
+                int oneDayInStudy = calculateDaysSince(oneStartTime.getValue());
+                dayInStudyMap.put(oneStartTime.getKey(), oneDayInStudy);
+            }
+            record.setDayInEachStudy(dayInStudyMap);
+
         } else {
             // default sharing to NO_SHARING
             record.setUserSharingScope(SharingScope.NO_SHARING);
         }
     }
+
+    protected int calculateDaysSince(DateTime studyStartTime) {
+        LocalDate studyStartDate = studyStartTime.withZone(BridgeConstants.LOCAL_TIME_ZONE).toLocalDate();
+        LocalDate todayLocalDate = DateUtils.getCurrentCalendarDateInLocalTime();
```

Comment:
So originally, we made the design decision to use the current time to calculate dayInStudy. However, this has the side effect that if we redrive the upload on a later date, the dayInStudy may change. This is probably not desirable behavior. Consider changing this to use record.getCreatedOn() instead. (NOTE: This is guaranteed to exist by UploadFormatHandler, and can be overwritten by IosSchemaValidationHandler2 and GenericUploadFormatHandler as they parse the upload.)

Also, while we're at it, instead of using BridgeConstants.LOCAL_TIME_ZONE, we can use record.getCreatedOnTimeZone() to get the participant's local time zone. (And fall back to BridgeConstants.LOCAL_TIME_ZONE if it doesn't exist.)